src/libs/ck-libs/io/ckio.C

```diff
@@ -138,7 +138,7 @@ namespace Ck { namespace IO {
             opts.skipPEs = CkMyNodeSize();
 	  if(opts.numReaders == 0){
 		// CkPrintf("DEBUG: the num readers in opts struct is 0\n");
-		opts.numReaders = std::min(32, CkNumPes());
+		opts.numReaders = std::min(CmiNumNodes(), CkNumPes());
```

Comment:
I don't know if it's possible for the number of PEs to be less than the number of nodes. You might be able to make this just `CkNumNodes()`.